java.source.base/src/org/netbeans/modules/java/source/save/CasualDiff.java

```diff
@@ -1477,27 +1477,26 @@ private int diffVarDef(JCVariableDecl oldT, JCVariableDecl newT, int[] bounds) {
                     int modsUpperBound = getCommentCorrectedEndPos(oldT.mods);
                     if (modsUpperBound > -1) {
                         tokenSequence.move(modsUpperBound);
-                        tokenSequence.moveNext();
 
                         // copying modifiers from oldTree
-                        if (JavaTokenId.WHITESPACE == tokenSequence.token().id()) {
-                            int offset = tokenSequence.offset();
-                            copyTo(localPointer, localPointer = offset);
-                        } else {
+                        if (tokenSequence.moveNext()) {
```

Comment:
The former code in commit #96003dd59aa2003dcbf086b231f8a5d5403e818a checked for `JavaTokenId.WHITESPACE` at `modsUpperBound`, and if there was whitespace did not copy it over. This update changes that behaviour, copying up to `modsUpperBound`. 

If that change is OK, then the tokenSequence work on lines 1479-1482 is probably useless, the sequence is positioned, but no token is consumed - and regardless of execution branch taken, the `tokenSequence` is always repositioned to a new place at line 1489.

Reply:
check for JavaTokenId.WHITESPACE at modsUpperBound was removed in subsequent commit as check was not required after the code refactoring.
Plz consider below scenario.

Scenario1:
/*comment1*/var/*comment2*/ a = {2,3.1f};

For above  case  getCommentCorrectedEndPos(oldT.mods) will be -1 .
So  Netbeans will not run code  inside the check (modsUpperBound > -1) present at line 1478
In that case it will loop at line 1492 to add all the whitespaces/comments.

Scenario2:
@NotNull var j = {new Object(),new Object()};

in this case code inside getCommentCorrectedEndPos(oldT.mods) >1 check 
will add '@NotNull' to the final statement string.

But we need to add the whitespace present after '@NotNull' to the final string.
For that reason while loop at line 1492 is required. 
it loops at line 1492 to add back the whitespace.

Scenario3:
final/*comment1*/var /*comment2*/ a = {2,3.1f};

check (modsUpperBound > -1) at line 1478 will add 'final/*comment1*/' to the final string. 
If this scenario modsUpperBound offset is pointing to VAR token.
As we want to skip adding VAR token in final String tokenSequence.move(localPointer) is required at line 1489.
Otherwise it will add wrongly 'var /*comment2*/' to the final string after looping at line 1492